src/game/Spell.cpp

```diff
@@ -6481,10 +6481,11 @@ bool Spell::CheckTarget(Unit* target, SpellEffectIndex eff) const
             break;
         default:                                            // normal case
             // Get GO cast coordinates if original caster -> GO
-            if (target != m_caster)
-                if (WorldObject* caster = GetCastingObject())
-                    if (!target->IsWithinLOSInMap(caster))
-                        return false;
+            if (!IsIgnoreLosSpell(m_spellInfo))
+                if (target != m_caster)
+                    if (WorldObject* caster = GetCastingObject())
+                        if (!target->IsWithinLOSInMap(caster))
```

Comment:
It's nice when things are readable too, also how would you make one line out of it when there's an assignment in there? It'd require at least 2 lines?